src/Services/Masa.Auth.Service.Admin/Domain/Permissions/EventHandler/UpdateRoleLimitDomainEventHandler.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Masa.Auth.Service.Admin.Domain.Permissions.Events;
+
+namespace Masa.Auth.Service.Admin.Domain.Permissions.EventHandler
+{
+    public class UpdateRoleLimitDomainEventHandler
+    {
+        readonly IRoleRepository _roleRepository;
+        readonly AuthDbContext _authDbContext;
+
+        public UpdateRoleLimitDomainEventHandler(IRoleRepository roleRepository, AuthDbContext authDbContext)
+        {
+            _roleRepository = roleRepository;
+            _authDbContext = authDbContext;
+        }
+
+        [EventHandler(1)]
+        public async Task UpdateRoleLimitAsync(UpdateRoleLimitDomainEvent roleEvent)
+        {
+            var roles = await _authDbContext.Set<Role>()
+                                    .Include(r => r.Users)
+                                    .Include(r => r.Teams)
+                                    .ThenInclude(tr => tr.Team)
+                                    .ThenInclude(t => t.TeamStaffs)
+                                    .Where(r => roleEvent.Roles.Contains(r.Id))
+                                    .ToListAsync();
+            foreach (var role in roles)
+            {
+                var quantityAvailable = GetQuantityAvailable(role);
+                if (role.Limit == 0 || quantityAvailable >= 0)
```

Comment:
quantityAvailable = 0?

Reply:
quantityAvailable = 0 means the character cardinality constraint has been used up